Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -0,0 +1,647 @@
+                    list.Add((uid, xform));
+
+                return;
+            }
+
+            // Not parented to map. Likely in a container.
+            if (_containerSystem.IsEntityInContainer(uid, xform))
+                return;
+
+            // finally check if it intersects our tile
+            if (gridBox.Contains(invSpaceMatrix.Transform(xform.LocalPosition)))
+                list.Add((uid, xform));
+        };
+
+        _entityLookup.FastEntitiesIntersecting(lookup, ref worldBox, callback);
+
+        foreach (var (entity, xform) in list)
+        {
+            processed.Add(entity);
+            ProcessEntity(entity, epicenter, damage, throwForce, id, damageQuery, physicsQuery, xform);
+        }
+
+        if (throwForce <= 0)
+            return;
+
+        // Also, throw any entities that were spawned as shrapnel. Compared to entity spawning & destruction, this extra
+        // lookup is relatively minor computational cost, and throwing is disabled for nukes anyways.
+        list.Clear();
+        _entityLookup.FastEntitiesIntersecting(lookup, ref worldBox, callback);
+        foreach (var (entity, xform) in list)
+        {
+            ProcessEntity(entity, epicenter, null, throwForce, id, damageQuery, physicsQuery, xform);
+        }
+    }
+
+    /// <summary>
+    ///     This function actually applies the explosion affects to an entity.
+    /// </summary>
+    private void ProcessEntity(
+        EntityUid uid,
+        MapCoordinates epicenter,
+        DamageSpecifier? damage,
+        float throwForce,
+        string id,
+        EntityQuery<DamageableComponent> damageQuery,
+        EntityQuery<PhysicsComponent> physicsQuery,
+        TransformComponent? xform = null)
+    {
+        // damage
+        if (damage != null && damageQuery.TryGetComponent(uid, out var damageable))
+        {
+            var ev = new GetExplosionResistanceEvent(id);
+            RaiseLocalEvent(uid, ev, false);
+
+            if (ev.Resistance == 0)
+            {
+                // no damage-dict multiplication required.
+                _damageableSystem.TryChangeDamage(uid, damage, ignoreResistances: true, damageable: damageable);
+            }
+            else if (ev.Resistance < 1)
+            {
+                _damageableSystem.TryChangeDamage(uid, damage * (1 - ev.Resistance), ignoreResistances: true, damageable: damageable);
+            }
+        }
+
+        // throw
+        if (xform != null
+            && !xform.Anchored 
+            && throwForce > 0
+            && !EntityManager.IsQueuedForDeletion(uid)
+            && physicsQuery.TryGetComponent(uid, out var physics)
+            && physics.BodyType == BodyType.Dynamic)
+        {
+            // TODO purge throw helpers and pass in physics component
+            uid.TryThrow(xform.WorldPosition - epicenter.Position, throwForce);
+        }
+
+        // TODO EXPLOSION puddle / flammable ignite?
```

Comment:
it should probably just apply temperature and flammable, similar to the hotspot effects in SS13. would be nice if i could use those for that purpose since CLF3-splosions are intended to use it but just do a small explosion rn

Reply:
Yeah I'll play around with `HotspotExpose`, but I'll leave that for a follow up PR because I imagine that will take some tweaking.